app/models/slot_availability.rb

```diff
@@ -3,30 +3,10 @@ class SlotAvailability
 
   def initialize(prison:, use_nomis_slots: false)
     @prison = prison
```

Comment:
Great! It's an interesting point r.e. `@prison`, but I think it's best as is. The instance variable is set for the `restrict_by_prisoner` method, which is called externally, however passing the prison parameter to the private methods makes the code simpler since it makes explicitly that they do not depend on mutable state.